### ISSUE TITLE
Cosmos: `serde_json::RawValue` fails to Deserialize on the Binary Path

### ISSUE THREAD

**Issue by kundadebdatta** [Azure/azure-sdk-for-rust#5328]
## Summary

Deserializing a response into `serde_json::value::RawValue` works on the text path and **fails on the binary path** with an opaque, misclassified error.

Because binary encoding is now **enabled by default**, this is a behavioral regression for any caller that used `RawValue` for raw document passthrough: code that worked before the default flip now fails at runtime with no API change on the caller's side.

## Observed failure (live account, Cosmos `read_item`)

```text
Error: 500/20020 (SerializationResponseBodyInvalid): failed to deserialize response body

Caused by:
  0: Custom("invalid type: newtype struct, expected any valid JSON value")
```

Note the diagnostics on that same failure: the HTTP request **succeeded**.

```json
{"status":"200","request_charge":1.0,"duration_ms":43,"timed_out":false}
```

So a `500`-class `SerializationResponseBodyInvalid` is reported for a perfectly valid `200` response. Nothing is wrong with the service payload — the client simply cannot deserialize it into the requested type under this encoding.

## Minimal offline repro

No account required:

```rust
use serde_json::value::RawValue;

let text = br#"{"v":1}"#;

let from_text: Result<Box<RawValue>, _> = serde_json::from_slice(text);
// Ok("{\"v\":1}")

let binary = azure_data_cosmos_driver::binary_json::transcode_to_binary(text).unwrap();
let from_binary: Result<Box<RawValue>, _> =
    azure_data_cosmos_driver::binary_json::from_slice(&binary);
// Err(Custom("invalid type: newtype struct, expected any valid JSON value"))
```

Actual output:

```text
text   -> RawValue: Ok("{\"v\":1}")
binary -> RawValue: Err(Custom("invalid type: newtype struct, expected any valid JSON value"))
```

## Root cause

`serde_json` implements `RawValue` with a magic newtype-struct name, `$serde_json::private::RawValue`, whose visitor expects the raw JSON **text**. A deserializer must recognize that token and supply it.

The **serializer already recognizes it** and deliberately rejects it with a clear, actionable message (`binary_json/ser.rs`):

```rust
const RAW_VALUE_TOKEN: &str = "$serde_json::private::RawValue";
// ...
if name == RAW_VALUE_TOKEN {
    return Err(BinaryError::Custom(
        "serde_json RawValue is not supported by the Cosmos binary JSON serializer".to_owned(),
    ));
}
```

The **deserializer has no counterpart** (`binary_json/de.rs`):

```rust
fn deserialize_newtype_struct<V>(self, _name: &'static str, visitor: V) -> Result<V::Value> {
    // Newtype structs serialize transparently, so deserialize the inner
    // value straight through.
    visitor.visit_newtype_struct(self)
}
```

`_name` is discarded, so the `RawValue` token is treated as an ordinary newtype struct. `visit_newtype_struct` is then rejected by `RawValue`'s visitor, which surfaces as the serde-internal `invalid type: newtype struct` message.

The serializer's own handling proves the token is known to this codec — the read side just never got the same treatment.

## Why this was not caught earlier

The driver uses `RawValue` extensively (`EnvelopeRow::payload` in the cross-partition `ORDER BY` pipeline), but **never through the binary deserializer**. `parse_envelope_page` transcodes the page to text first:

```rust
let bytes = normalize_page_body(&bytes)?;                 // binary -> TEXT
let feed: RawFeedBody = serde_json::from_slice(&bytes)?;  // RawValue parsed from TEXT
```

So internal `RawValue` parsing is always handled by `serde_json`, never by `binary_json::from_slice`.

The caller's path differs: `into_model` / `into_items` detect a binary payload and route it **straight** into `binary_json::from_slice` with no text hop — the only path that reaches the broken branch.

Consequence worth noting: the redundant transcode that the "binary-aware `parse_envelope_page`" work aims to remove is exactly what masks this defect today. Removing it without fixing this first turns a caller-facing bug into a driver-internal one.

Secondary reasons it stayed hidden:

- No test deserializes into `RawValue` from a binary payload. The round-trip fuzzers and corpus tests use `serde_json::Value`; the typed probe is a plain struct. Neither triggers `deserialize_newtype_struct`.
- Binary encoding was opt-in until recently, so a `RawValue` caller would not have hit it before the default flip.
- It requires an uncommon combination: binary enabled **and** the caller requesting `RawValue`. Typed models and `Value` both work.

## Impact

- **Regression on upgrade.** Binary is on by default, so `RawValue` callers break without changing any code.
- **Raw passthrough is a real pattern.** Migration tools, proxies, and audit/diff utilities use `RawValue` precisely to avoid re-serializing documents. It is also the only way to observe a document's exact bytes, since `serde_json::Value` normalizes key order and number spelling.
- **Misleading diagnosis.** The error is classified as a *service response body* problem (`SerializationResponseBodyInvalid`, 500-class) on an HTTP `200`, which sends operators looking at the service instead of the encoding.
- **Silent discovery cost.** The failure appears only at runtime, on the read path, after a successful round trip to the service.

## Suggested fix

Two tiers; the first is the minimum bar, the second is the real fix.

### Tier 1 — make the failure honest (small)

- Match `RAW_VALUE_TOKEN` in `deserialize_newtype_struct` and return the same clear message shape the serializer uses, rather than letting a serde-internal type error escape.
- Reclassify: this is a client-side unsupported-type condition, not an invalid service body. A `200` response should not produce a 500-class `SerializationResponseBodyInvalid`.
- Document the limitation on the public entry points (`ResponseBody::into_model` / `into_items`) and in the binary-encoding spec.

### Tier 2 — actually support it (medium)

On read, `RawValue` is satisfiable: transcode the value at the cursor to text and hand the visitor that string. `transcode_to_text` already exists, and `Reader::new(buf, pos)` can read from an arbitrary offset, so a sub-value can be rendered without decoding the whole payload.

> :warning: Do **not** implement this by slicing the binary buffer and re-prefixing `0x80`. Reference strings (`STR_R1`..`STR_R4`) resolve against **absolute page offsets**, so a detached sub-slice mis-resolves any reference pointing outside it — and does so **silently**, returning wrong text rather than erroring. Render through a reader positioned on the full buffer.

For the write side, the serializer's current rejection is defensible, but the same transcode-based approach (`transcode_to_binary` on the raw JSON text) would make the type work symmetrically.

## Acceptance criteria

- [ ] `binary_json::from_slice::<Box<RawValue>>(..)` either succeeds with the value's JSON text, or fails with a message that names `RawValue` and the binary path.
- [ ] A `200` response never yields a 500-class `SerializationResponseBodyInvalid` for an unsupported caller type.
- [ ] `into_model::<Box<RawValue>>()` behaves identically under text and binary, or the difference is documented on the public API.
- [ ] Regression test covering both encodings, including a payload containing reference strings whose targets lie outside the extracted value.
- [ ] Changelog entry, since this is caller-visible behavior.

## Notes

Found while investigating a reported wide-integer precision issue. That investigation cleared binary encoding of the precision fault — the service stores numbers as IEEE-754 doubles, so a 48-digit integer collapses on the pure text path too — but the `RawValue` read failure surfaced along the way and is unrelated to it.
